shell/platform/common/cpp/client_wrapper/plugin_registrar.cc

```diff
@@ -87,6 +93,27 @@ void BinaryMessengerImpl::Send(const std::string& channel,
                               message_size);
 }
 
+void BinaryMessengerImpl::Send(const std::string& channel,
+                               const uint8_t* message,
+                               const size_t message_size,
+                               BinaryReply reply) const {
+  struct Captures {
+    BinaryReply reply;
+  };
+  auto captures = new Captures();
+  captures->reply = reply;
+
+  auto message_reply = [](const uint8_t* data, size_t data_size,
+                          void* user_data) {
+    auto captures = reinterpret_cast<Captures*>(user_data);
+    captures->reply(data, data_size);
+    delete captures;
+  };
+  FlutterDesktopMessengerSendWithReply(messenger_, channel.c_str(), message,
+                                       message_size, std::move(message_reply),
```

Comment:
I'm not following the lifetime/ownership here. Why is an API function pointer being passed something via `std::move`? What is guaranteeing the lifetime of the closure here. (Is it safe because there's no capturing? I forget the exact rules about this for C++.)

Reply:
I forgot to remove. I was getting a crash on this API and initially thought it was related to this. Removed